lib/jekyll/admin/server/collection.rb

```diff
@@ -66,20 +66,25 @@ def ensure_document
           render_404 if document.nil?
         end
 
-        # Returns the document without front matter defaults by
-        # Recalculating @data using a modified version of Document#read,
-        # but without first setting the defaults
-        def document_without_front_matter_defaults(original_doc = document)
+        # Returns a document suitable for to_liquid being called on for the API
+        # The document will be the document as seen by Jekyll, but:
+        #  1. without front matter defaults by
+        #  2. With a `raw_content` field added
+        def document_for_api(original_doc = document)
           doc = original_doc.dup
           doc.instance_variable_set "@data", {}
 
           # If the file has YAML front matter, read it in
           content = File.read(doc.path, Utils.merged_file_read_opts(site, {}))
           if content =~ Jekyll::Document::YAML_FRONT_MATTER_REGEXP
+            content = $POSTMATCH
             data_file = SafeYAML.load(Regexp.last_match(1))
             doc.merge_data!(data_file, :source => "YAML front matter") if data_file
           end
 
+          # Return the unrendered markdown, see https://git.io/vKDTG
+          doc.merge_data!({ "raw_content" => content }, :source => "Jekyll Admin")
```

Comment:
We should add this field to `Document` and `Page` in Jekyll proper eventually. So dumb that we don't. `raw_yaml`, too, maybe?